src/main/java/it/plansoft/classmanager/model/Classroom.java

```diff
@@ -57,4 +57,40 @@ public int getGrade() {
 	public void setStudents(Set<Student> students) {
 		this.students = students;
 	}
+
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj) return true;
+		if (obj == null) return false;
+		if (getClass() != obj.getClass()) return false;
+		
+		Classroom other = (Classroom) obj;
+		if (grade != other.grade) return false;
+		if (id == null) {
```

Comment:
Perchè hai fatto un if annidato invece che un solo if?

```suggestion
		if (id == null) && (other.id != null) {
```

Reply:
Come anche negli altri if che ci sono

Reply:
Sono metodi autogenerati attraverso Eclipse, provvederò a formattarli meglio.

Reply:
Quando risolvi un commento ricordati di cliccare su "Resolve conversation"